pkg/db/db.go

```diff
@@ -4309,3 +4344,443 @@ func (h *DBHandler) RunCustomMigrationEnvironments(ctx context.Context, getAllEn
+	}
+	if app.Locks == nil {
+		app.Locks = map[string]*api.Lock{}
+	}
+	app.Locks[applicationLock.LockID] = &api.Lock{
+		Message:   applicationLock.Metadata.Message,
+		LockId:    applicationLock.LockID,
+		CreatedAt: timestamppb.New(applicationLock.Created),
+		CreatedBy: &api.Actor{
+			Name:  applicationLock.Metadata.CreatedByName,
+			Email: applicationLock.Metadata.CreatedByEmail,
+		},
+	}
+	if applicationLock.Deleted {
+		delete(app.Locks, applicationLock.LockID)
+	}
+	err = h.WriteOverviewCache(ctx, transaction, latestOverview)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *DBHandler) UpdateOverviewRelease(ctx context.Context, transaction *sql.Tx, release DBReleaseWithMetaData) error {
+	latestOverview, err := h.ReadLatestOverviewCache(ctx, transaction)
+	if err != nil {
+		return err
+	}
+	if h.CheckOverviewIsEmpty(latestOverview) {
+		return nil
+	}
+	app := getApplicationByName(latestOverview.Applications, release.App)
+	if app == nil {
+		return fmt.Errorf("could not find application %s in overview", release.App)
+	}
+	apiRelease := &api.Release{
+		PrNumber:        extractPrNumber(release.Metadata.SourceMessage),
+		Version:         release.ReleaseNumber,
+		UndeployVersion: false,
+		SourceAuthor:    release.Metadata.SourceAuthor,
+		SourceCommitId:  release.Metadata.SourceCommitId,
+		SourceMessage:   release.Metadata.SourceMessage,
+		CreatedAt:       timestamppb.New(release.Created),
+		DisplayVersion:  release.Metadata.DisplayVersion,
+	}
+	foundRelease := false
+	for relIndex, currentRelease := range app.Releases {
+		if currentRelease.Version == release.ReleaseNumber {
+			if release.Deleted {
+				app.Releases = append(app.Releases[:relIndex], app.Releases[relIndex+1:]...)
+			} else {
+				app.Releases[relIndex] = apiRelease
+			}
+			foundRelease = true
+		}
+	}
+	if !foundRelease && !release.Deleted {
+		app.Releases = append(app.Releases, apiRelease)
+	}
+	err = h.WriteOverviewCache(ctx, transaction, latestOverview)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *DBHandler) ForceOverviewRecalculation(ctx context.Context, transaction *sql.Tx) error {
+	emptyOverview := &api.GetOverviewResponse{
+		Applications:      map[string]*api.Application{},
+		EnvironmentGroups: []*api.EnvironmentGroup{},
+		GitRevision:       "",
+		Branch:            "",
+		ManifestRepoUrl:   "",
+	}
+	err := h.WriteOverviewCache(ctx, transaction, emptyOverview)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *DBHandler) CheckOverviewIsEmpty(overviewResp *api.GetOverviewResponse) bool {
+	if overviewResp == nil {
+		return true
+	}
+	if len(overviewResp.Applications) == 0 && len(overviewResp.EnvironmentGroups) == 0 && overviewResp.GitRevision == "" {
+		return true
+	}
+	return false
+}
+
+func getEnvironmentByName(groups []*api.EnvironmentGroup, envNameToReturn string) *api.Environment {
```

Comment:
It's a bit surprising to find these functions here.
Maybe if we just move everything here into db/overview.go it would be a bit more logical.